File: Youtube_python/video_bd.py
from video import Video

class VideoBd:
    def __init__(self):
        self.video_list = []
    
    def add_video(self, video):
        self.video_list.append(video)
    
    def research_video(self, nom):
        for v in self.video_list:
            if v.titre_video == nom:
                print(f"\t \tID Vidéo : {v.video_id}")
                print(f"\t \tTitre : {v.titre_video}")
                print(f"\t \tAutheur : {v.author}")
                print(f"\t \tDescription : {v.description}")
                print(f"\t \tResolution : {v.resolution}K")
                print(f"\t \tDate de Publication : {v.date_publication}")
                print("\t \t----------------------------------------")
                print("\t \tSTATISTIQUES & PERTINENCE\n")
                print(f"\t \tNombre de Vues : {v.nbre_view}")
                print(f"\t \tNombre de Commentaires : {v.nbre_comment}")
                print(f"\t \tNombre de Likes : {v.nbre_likes}")
                print(f"\t \tScore de Pertinence : {v.get_relevance_score()}")
                print(f"\t \tStatut : {v.get_pertinence_status()}")
                print("\t \t----------------------------------------\n\n")
                v.display_list_comments()
                return
        print("Aucune video disponible pour ce theme.....")
    
    def liked_video(self, titre):
        for v in self.video_list:
            if v.titre_video == titre:
                v.nbre_likes += 1
                print(f"La vidéo '{titre}' a été likée ! Nouveau nombre de likes : {v.nbre_likes}")
                return
        print(f"Vidéo introuvable : {titre}")
    
    def get_video_by_title(self, titre):
        for v in self.video_list:
            if v.titre_video == titre:
                return v
        return None
    
    def get_video_by_id(self, video_id):
        for v in self.video_list:
            if v.video_id == video_id:
                return v
        return None
    
    def display_video(self):
        if not self.video_list:
            print("Aucune vidéo disponible.")
            return
        
        for v in self.video_list:
            print("\n========================================")
            print(f"ID : {v.video_id}")
            print(f"Titre de la Video : {v.titre_video}")
            print(f"Author: {v.author}")
            print(f"Description: {v.description}")
            print(f"Resolution: {v.resolution}K")
            print(f"Date: {v.date_publication}")
            print(f"Nbre comment: {v.nbre_comment}")
            print(f"Nbre vue: {v.nbre_view}")
            print(f"Likes: {v.nbre_likes}")
            print("========================================\n")
    
    def delete_video(self, nom):
        for v in self.video_list:
            if v.titre_video == nom:
                self.video_list.remove(v)
                print(f"Vidéo '{nom}' supprimée avec succès.")
                return
        print(f"Vidéo introuvable : {nom}")

File: Youtube_c++/VideoBd.cpp
#include "VideoBd.h"
#include <iostream>

VideoBd::VideoBd() {}

void VideoBd::addVideo(const Video& v) {
    videoList.push_back(v);
}

void VideoBd::researchVideo(const std::string& nom) {
    for (const auto& v : videoList) {
        if (v.titreVideo == nom) {
            std::cout << "\t \tID Vidéo : " << v.videoId << "\n";
            std::cout << "\t \tTitre : " << v.titreVideo << "\n";
            std::cout << "\t \tAutheur : " << v.author << "\n";
            std::cout << "\t \tDescription : " << v.description << "\n";
            std::cout << "\t \tResolution : " << v.resolution << "K\n";
            std::cout << "\t \tDate de Publication : " << v.datePublication << "\n";
            std::cout << "\t \t----------------------------------------\n";
            std::cout << "\t \tSTATISTIQUES & PERTINENCE\n\n";
            std::cout << "\t \tNombre de Vues : " << v.nbreView << "\n";
            std::cout << "\t \tNombre de Commentaires : " << v.nbreComment << "\n";
            std::cout << "\t \tNombre de Likes : " << v.nbreLikes << "\n";
            std::cout << "\t \tScore de Pertinence : " << v.getRelevanceScore() << "\n";
            std::cout << "\t \tStatut : " << v.getPertinenceStatus() << "\n";
            std::cout << "\t \t----------------------------------------\n\n";
            v.displayListComments();
            return;
        }
    }
    std::cout << "Aucune video disponible pour ce theme.....\n";
}

void VideoBd::likedVideo(const std::string& titre) {
    for (auto& v : videoList) {
        if (v.titreVideo == titre) {
            v.nbreLikes++;
            std::cout << "La vidéo '" << titre << "' a été likée ! Nouveau nombre de likes : " << v.nbreLikes << "\n";
            return;
        }
    }
    std::cout << "Vidéo introuvable : " << titre << "\n";
}

Video* VideoBd::getVideoByTitle(const std::string& titre) {
    for (auto& v : videoList) {
        if (v.titreVideo == titre) {
            return &v;
        }
    }
    return nullptr;
}

Video* VideoBd::getVideoById(const std::string& videoId) {
    for (auto& v : videoList) {
        if (v.videoId == videoId) {
            return &v;
        }
    }
    return nullptr;
}

void VideoBd::displayVideo() const {
    if (videoList.empty()) {
        std::cout << "Aucune vidéo disponible.\n";
        return;
    }
    
    for (const auto& v : videoList) {
        std::cout << "\n========================================\n";
        std::cout << "ID : " << v.videoId << "\n";
        std::cout << "Titre de la Video : " << v.titreVideo << "\n";
        std::cout << "Author: " << v.author << "\n";
        std::cout << "Description: " << v.description << "\n";
        std::cout << "Resolution: " << v.resolution << "K\n";
        std::cout << "Date: " << v.datePublication << "\n";
        std::cout << "Nbre comment: " << v.nbreComment << "\n";
        std::cout << "Nbre vue: " << v.nbreView << "\n";
        std::cout << "Likes: " << v.nbreLikes << "\n";
        std::cout << "========================================\n";
    }
}

void VideoBd::deleteVideo(const std::string& nom) {
    for (auto it = videoList.begin(); it != videoList.end(); ++it) {
        if (it->titreVideo == nom) {
            videoList.erase(it);
            std::cout << "Vidéo '" << nom << "' supprimée avec succès.\n";
            return;
        }
    }
    std::cout << "Vidéo introuvable : " << nom << "\n";
}